packages/spark-core/base/ssnInput.js

```diff
@@ -0,0 +1,45 @@
+import {toggleValue} from '../utilities/toggleValue';
+
+function ssnInput () {
+  const ssnInputs = document.querySelectorAll('[data-sprk-input="ssn"]');
+
+  ssnInputs.forEach(ssnInput => {
+    const field = ssnInput.querySelectorAll('input')[0];
+    const checkbox = ssnInput.querySelectorAll('input[type=checkbox]')[0];
+
+    // set up the 'show ssn' button
+    checkbox.addEventListener('change', () => {
+      field.setAttribute('type', toggleValue(field.getAttribute('type'), 'text', 'password'));
+    });
+
+    // set up the autoformat
+    field.addEventListener('blur', () => {
+      field.value = formatSSN(field);
+    });
+  });
+}
+
+function formatSSN (element) {
+  let val = element.value;
+  let newVal = '';
+  const sizes = [3, 2, 4];
+
+  if (!isNaN(val) && val.length === 9) {
```

Comment:
isNaN() has some interesting results like it will return false in the case of a null value `isNaN(null)` or empty string `isNaN('')`.  Which, we wouldn't want to happen. But, since you're also checking the length that would rule out those two scenarios. But, back to the "interesting results" bit that I mentioned. This will also return false `isNaN('0xF111111')`. isNaN thinks that '0xF111111' is a number. Which, results in those 9 chars passing your if statement isNaN and length check and getting formatted. We might want to add an enhanced check when we do get to breaking this out to a validator. 
<img width="480" alt="screen shot 2018-03-13 at 9 11 00 pm" src="https://user-images.githubusercontent.com/2117593/37377946-66b97f1a-2703-11e8-8fc3-0263f14ec24a.png">